src/cast/safe/sign.rs

```diff
@@ -0,0 +1,232 @@
+/// Casts an integer to a different sign without changing its size.
+///
+/// `SignCast` exposes the cast operator in generic contexts. It is used to
+/// perform a signed cast between integers of the same size. The `SignCast`
+/// trait will **never** change the size of the integer.
+pub trait SignCast {
+    type Unsigned;
+    type Signed;
+
+    fn unsigned(self) -> Self::Unsigned;
+    fn signed(self) -> Self::Signed;
+}
```

Comment:
I do like this trait -- I have often wanted this functionality.

Oddly, this one actually is a bitwise conversion, but you didn't call it such...

I think we should strengthen the associated types using the sign traits:

```rust
    type Unsigned: Unsigned;
    type Signed: Signed;
```

Reply:
That makes sense to me.